feat(webvitals): More updates to Webvitals module components to use backend scores

edwardgou-sentry opened the pull request:
Updates more webvitals components to use backend scores:
- Updates more queries to use `performance_score(...)` and `weighted_performance_score(...)` functions
- Updates performance score breakdown chart to support using backend scores
- Reorganizes a few queries
- Update `calculatePerformanceScoreFromStored.tsx` to utilize properly `performance_score(...)` function
- Clean up some unused query fields
- Cleans up `orderby` from some queries, due to being replaced by sort hook

Backend score usage is still controlled by the `USE_STORED_SCORES` constant, which is hardcoded to false for the moment.

edwardgou-sentry committed:
Refactors webvitals queries to support querying for stored webvital scores

edwardgou-sentry committed:
referrer and enabled

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into egou/feat/webvitals-backend-scores-support-2

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into egou/feat/webvitals-backend-scores-support-2

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
progress

edwardgou-sentry committed:
settings

edwardgou-sentry committed:
show full series and only query samples with scores

edwardgou-sentry committed:
use metrics

gggritso commented on static/app/views/performance/browser/webVitals/utils/queries/rawWebVitalsQueries/useProjectRawWebVitalsTimeseriesQuery.tsx:
```diff
@@ -0,0 +1,142 @@
+  };
+
+  result?.data?.['p75(measurements.lcp)']?.data.forEach((interval, index) => {
+    const lcp: number = result?.data?.['p75(measurements.lcp)']?.data[index][1][0].count;
+    const fcp: number = result?.data?.['p75(measurements.fcp)']?.data[index][1][0].count;
+    const cls: number = result?.data?.['p75(measurements.cls)']?.data[index][1][0].count;
+    const ttfb: number =
+      result?.data?.['p75(measurements.ttfb)']?.data[index][1][0].count;
+    const fid: number = result?.data?.['p75(measurements.fid)']?.data[index][1][0].count;
+    // This is kinda jank, but since events-stats zero fills, we need to assume that 0 values mean no data.
+    // 0 value for a webvital is low frequency, but not impossible. We may need to figure out a better way to handle this in the future.
```
That's an awful lot of repetition! Iterating though `['lcp', 'fcp', 'cls', 'ttfb', 'fid']` would be way nicer

gggritso replied:
Also applies to all the code below

edwardgou-sentry replied:
I've cleaned this up to be much shorter now!

gggritso commented on static/app/views/performance/browser/webVitals/utils/queries/storedScoreQueries/calculatePerformanceScoreFromStored.tsx:
```diff
@@ -28,21 +28,12 @@ export function getWebVitalScores(data?: TableDataRow): ProjectScore {
-    fidScore: Math.round(
-      getWebVitalScore(data, 'fid') * getWebVitalScoreWeight(data, 'fid') * 100
-    ),
+    lcpScore: Math.round(getWebVitalScore(data, 'lcp') * 100),
+    fcpScore: Math.round(getWebVitalScore(data, 'fcp') * 100),
+    clsScore: Math.round(getWebVitalScore(data, 'cls') * 100),
+    ttfbScore: Math.round(getWebVitalScore(data, 'ttfb') * 100),
+    fidScore: Math.round(getWebVitalScore(data, 'fid') * 100),
+    totalScore: Math.round(getTotalScore(data) * 100),
```
These don't have to use weights?

edwardgou-sentry replied:
This function is supposed to get you the individual 0 - 100 webvital scores (used to display in the tile components) and total score, so we don't need to apply weights here

gggritso commented on static/app/views/performance/browser/webVitals/utils/queries/storedScoreQueries/useProjectWebVitalsScoresTimeseriesQuery.tsx:
```diff
@@ -0,0 +1,143 @@
+
+  result?.data?.['weighted_performance_score(measurements.score.lcp)']?.data.forEach(
+    (interval, index) => {
+      const lcp: number =
+        result?.data?.['weighted_performance_score(measurements.score.lcp)']?.data[
+          index
+        ][1][0].count * 100;
+      const fcp: number =
+        result?.data?.['weighted_performance_score(measurements.score.fcp)']?.data[
+          index
+        ][1][0].count * 100;
+      const cls: number =
+        result?.data?.['weighted_performance_score(measurements.score.cls)']?.data[
+          index
+        ][1][0].count * 100;
+      const ttfb: number =
+        result?.data?.['weighted_performance_score(measurements.score.ttfb)']?.data[
+          index
+        ][1][0].count * 100;
+      const fid: number =
+        result?.data?.['weighted_performance_score(measurements.score.fid)']?.data[
+          index
+        ][1][0].count * 100;
```
Same note about repetition as before

edwardgou-sentry replied:
Cleaned up!

gggritso reviewed (approved):
Makes sense! Nits only

edwardgou-sentry committed:
Merge branch 'master' of github.com:getsentry/sentry into egou/feat/webvitals-backend-scores-support-2

edwardgou-sentry committed:
rename to preserveIncompletePoints and cleaned up lengthy code

edwardgou-sentry commented on static/app/views/starfish/components/chart.tsx:
```diff
@@ -86,6 +86,7 @@ type Props = {
   durationUnit?: number;
   errored?: boolean;
   forwardedRef?: RefObject<ReactEchartsRef>;
+  fullSeries?: boolean;
```
👍 renamed!

edwardgou-sentry committed:
feat(webvitals): Updates webvital values from p75 to avg (#61191)

Updates aggregate webvital values being displayed from p75 to avg. This
closer matches how performance scores are calculated and displayed.
Also adds a new tooltip to the webvitals landing tiles.

edwardgou-sentry committed:
fix merge
